tests/platform_tests/api/test_fan.py

```diff
@@ -54,6 +62,25 @@ def setup(self, platform_api_conn):
             except:
                 pytest.fail("num_fans is not an integer")
 
+    #
+    # Helper functions
+    #
+
+    def compare_value_with_platform_facts(self, key, value, fan_idx):
+        expected_value = None
+
+        if self.chassis_facts:
+            expected_fans = self.chassis_facts.get("fans")
+            if expected_fans:
+                expected_value = expected_fans[fan_idx].get(key)
+
+        if not expected_value:
+            logger.warning("Unable to get expected value for '{}' from platform.json file for fan {}".format(key, fan_idx))
+            return
```

Comment:
Remove warning message and make the check mandatory. See SFP portion of https://github.com/Azure/sonic-mgmt/pull/2268 for examples.